compat/mingw.c

```diff
@@ -482,7 +519,9 @@ int mingw_rmdir(const char *pathname)
 		return -1;
 
 	do {
-		if (!_wrmdir(wpathname))
+		int res = tries > 0 ? do_unlink(wpathname, _wrmdir)
```

Comment:
What is the reason for this change? Does `_wrmdir` fail if a file in the directory is open? If it's the same as above (directory cannot be recreated because of open file handles), then I think we would have to use rename and unlink also for the first iteration.

Reply:
I don't remember. I think I reached this case with a symbolic link to a directory (that was not even open).

Reply:
Without a valid use case it is impossible to tell if this change makes sense. Please revert. If this change is needed, please write a test to demonstrate the issue.